File: CSC-308/Ruby/BubbleSort.rb
# FVCproductions
# March 23, 2016
# Bubble Sort Program
#!/usr/bin/ruby

# define array

array = [ 34, 56, 4, 10, 77, 51, 93, 30, 5, 52 ]

# print out unsorted array

print "Unsorted Array \n"

for i in 0..(array.length - 2)
  print array[i], " "
end

# bubble sort algo

for i in 0..(array.length - 2)
  for j in i + 1..(array.length - 1)
    # swap method described in class
    if array[i] > array[j]
      array[i], array[j] = array[j], array[i]
    end
  end
end

# print sorted array

print "\n\nSorted Array\n"

for i in 0..array.length-2
  print array[i], " "
end

# press any key to continue

print "\n\nPress any key to continue . . ."
gets

File: CSC-308/Python/BubbleSort.py
# FVCproductions
# Python Bubble Sort Program
# CSC 308

def BubbleSort(array):
  # print unsorted array
  print("Unsorted Array")
  buffer = ""
  for i in range(len(array)):
    buffer = buffer + str(array[i]) + " "
  print(buffer)
  # bubble sort
  for i in range(len(array)-1, 0, -1):
    for j in range(i):
      if array[j] > array[j+1]:
        array[j], array[j + 1] = array[j + 1], array[j]
  # print sorted array
  print("\nSorted Array")
  buffer = ""
  for i in range(len(array)):
    buffer = buffer + str(array[i]) + " "
  print(buffer)
  # input message
  input("\nPress any key to continue . . .")

# calling BubbleSort function with given numbers
BubbleSort([ 34, 56, 4, 10, 77, 51, 93, 30, 5, 52 ])